benchmarks/encoder/perf.py

```diff
@@ -0,0 +1,63 @@
+import abc
+import time
+import typing as t
+
+import attr
+import pyperf
+import six
+from contextlib import contextmanager
+
+
+@attr.s
+class Variant:
+    name = attr.ib(type=str)
+    spec = attr.ib(type=t.Dict[str, t.Union[int, str, bool]], factory=dict)
+
+
+@attr.s
+class Scenario(six.with_metaclass(abc.ABCMeta)):
+    variant = attr.ib(type=Variant)
+    name = attr.ib(type=str, init=False)
+    spec = attr.ib(type=t.Dict[str, t.Union[t.Type[int], t.Type[str]]], init=False)
+
+    @abc.abstractmethod
+    @contextmanager
+    def run_ctx(self):
```

Comment:
:+1: nice - this will allow senarios to setup, yield and then teardown if need-be!